pkg/endpoint/events.go

```diff
@@ -179,3 +184,64 @@ func (ev *EndpointPolicyVisibilityEvent) Handle(res chan interface{}) {
 	}
 	return
 }
+
+// EndpointPolicyBandwidthEvent contains all fields necessary to update
+// the Pod's bandwidth policy.
+type EndpointPolicyBandwidthEvent struct {
+	ep     *Endpoint
+	annoCB AnnotationsResolverCB
+}
+
+// Handle handles the policy bandwidth update.
+func (ev *EndpointPolicyBandwidthEvent) Handle(res chan interface{}) {
+	var bps uint64
+
+	e := ev.ep
+	if err := e.lockAlive(); err != nil {
+		// If the endpoint is being deleted, we don't need to
+		// update its bandwidth policy.
+		res <- &EndpointRegenerationResult{
+			err: nil,
+		}
+		return
+	}
+	defer func() {
+		e.unlock()
+	}()
+
+	bandwidthEgress, err := ev.annoCB(e.K8sNamespace, e.K8sPodName)
+	if err != nil || !option.Config.EnableBandwidthManager {
+		res <- &EndpointRegenerationResult{
+			err: err,
+		}
+		return
+	}
+	if bandwidthEgress != "" {
+		bps, err = bandwidth.GetBytesPerSec(bandwidthEgress)
+		if err == nil {
+			err = (&bwmap.ThrottleBPFMap{}).Update(e.ID, bps)
```

Comment:
nit: Instead of creating a struct each time invoking `Update/Delete`, you could just create an instance of the struct in the `bwmap` pkg, and all then invoke the methods for it.

Reply:
This is an empty struct, I've actually copied this from ipmasq map code. ;)

Reply:
Removed.

Reply:
Yeah, in ipmasq we use the struct to be able to mock the ipmasq map in unit tests.